### DISCUSSION TITLE
SVG image convert not working

### DISCUSSION THREAD

**Discussion by amannagarro01** [ImageMagick/ImageMagick#7829] [General]
Attached sample SVG format image
[Zombie Carnival_500x500_SVG (1).zip](https://github.com/user-attachments/files/18058424/Zombie.Carnival_500x500_SVG.1.zip)

I used below ImageMagick command to generate different size of thumbnails/preview for this image, but it is not working-

convert image:gm convert +repage  /usr/local/tomcat/temp/6c9eacb07709420dbdf66eef29bb7f81-tmp-store.svg.thumbnail[0]  mpr:image -resize 3600x3600 -write /usr/local/tomcat/temp/d9108e6d-4f13-4d18-9df5-7af8d16cefb0_6ee13fd46ebe4371b04d9f2ca31e152b.master mpr:image -resize 800x800 -write /usr/local/tomcat/temp/d9108e6d-4f13-4d18-9df5-7af8d16cefb0_6ee13fd46ebe4371b04d9f2ca31e152b.800.jpg.png mpr:image -resize 640x640 -write /usr/local/tomcat/temp/d9108e6d-4f13-4d18-9df5-7af8d16cefb0_6ee13fd46ebe4371b04d9f2ca31e152b.640.jpg.png mpr:image -resize 500x500 -write /usr/local/tomcat/temp/d9108e6d-4f13-4d18-9df5-7af8d16cefb0_6ee13fd46ebe4371b04d9f2ca31e152b.500.jpg.png mpr:image -resize 320x320 -write /usr/local/tomcat/temp/d9108e6d-4f13-4d18-9df5-7af8d16cefb0_6ee13fd46ebe4371b04d9f2ca31e152b.320.jpg.png mpr:image -resize 240x240 -write /usr/local/tomcat/temp/d9108e6d-4f13-4d18-9df5-7af8d16cefb0_6ee13fd46ebe4371b04d9f2ca31e152b.240.jpg.png mpr:image -resize 100x100  /usr/local/tomcat/temp/d9108e6d-4f13-4d18-9df5-7af8d16cefb0_6ee13fd46ebe4371b04d9f2ca31e152b.100.jpg.png

Can I get suggestion why this command does't work for this image, while similar command is working fine for other sample images of same format.

**Comment by snibgo** [comment]
Your command starts with:
```
convert image:gm convert +repage ...
```
If your IM is v7, you should use "magick", not "convert" which is deprecated.

What is "image:gm"? Do you really have a file called "image:gm"?

What is the second "convert" doing?

Then you have an input file called "/usr/local/tomcat/temp/6c9eacb07709420dbdf66eef29bb7f81-tmp-store.svg.thumbnail", with a really strange extension "thumbnail".

Then you read "mpr:image", but you haven't written "mpr:image", so that will fail.

At that point, I give up. The command is junk.

**Reply by amannagarro01** [reply]
This command is confusing you, this is getting used to generate different thumbnail size previews.
You can simply try generating preview of the given file using-
gm convert Zombie Carnival_500x500_SVG (1).svg -resize 800x800 -write output.jpg

Still I am unable to generate thumbnail.

**Comment by snibgo** [comment]
This forum is for ImageMagick. "gm" is GraphicsMagick, which is a different system.

**Comment by csyfly** [comment]
It's really frustrating when ImageMagick fails to convert SVG. This tool [SVGtoPNG](https://svg-to-png.net/) might come in handy. It's free and doesn't require you to upload files. You can download the converted PNG images instantly, so you can give it a try.
